tests/lint/genDiff.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: UTF-8 -*-
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2019 NV Access Limited
+
+import subprocess
+from typing import Tuple, List
+
+_baseBranchPlaceholder = "<baseBranch>"
+_mergeBaseCommand = [
+	"git",
+	# merge-base is used to limit changes to only those that are new
+	# on HEAD.
+	"merge-base",
+	_baseBranchPlaceholder,  # this is the target branch used in a PR
+	"HEAD"
+]
+_mergeBasePlaceholder = "<mergeBase>"
+_diffCommand = [
+	"git", "diff",
+	# Only include changed lines (no context) in the diff. Otherwise
+	# developers may end up getting warnings for code adjacent to code they
+	# touched. This could result in very large change sets in order to get a
+	# clean build.
+	"-U0", _mergeBasePlaceholder
+	# We don't use triple dot syntax ('...') because it will not
+	# report changes in your working tree.
+]
+
+
+def substitutePlaceholder(

```

Comment:
Why don't you use python string formatting here? e.g.:

```
_mergeBaseCommand = "git merge-base {baseBranchPlaceholder,  # this is the target branch used in a PR} HEAD"
```

then, you could call format on `_mergeBaseCommand` to pass it the actual branch you want to fill in.
Or, is it really required to use a list here?

Honestly, I really feel a bit uncomfortable with looking up the placeHolderIndex like this.

Reply:
I did it this way to match the examples for `subprocess.check_output`. I also don't really like to have to look up the index, I wanted the commands at the beginning of the file to make it clear what the point was. I'll see if using string.format works.

Reply:
I decided to keep the args in list, but join them with space (so we don't have to remember to put a space after each argument manually), then format the string.